app/helpers/application_helper.rb

```diff
@@ -260,6 +244,18 @@ def view_to_url(view, parent = nil)
     end
   end
 
+  def class_to_db(cname, parent)
+    return unless cname
+    case
+    when cname == "OrchestrationStackOutput" then "outputs"
+    when cname == "OrchestrationStackParameter" then "parameters"
+    when cname == "OrchestrationStackResource" then "resources"
+    when cname == "SystemService" && parent.class.base_class.to_s.downcase == "host" then "host_services"
+    when cname == "SystemService" && parent.class.base_class.to_s.downcase == "vm"   then @lastaction
```

Comment:
why not `parent.kind_of?(Vm)` instead of `parent.class.base_class.to_s.downcase == "vm"`